src/Conversion/ONNXToKrnl/Math/Elementwise.cpp

```diff
@@ -331,6 +334,74 @@ struct ScalarOp<ONNXTanOp> {
   using IOp = NotSuportedScalarOp;
 };
 
+//===----------------------------------------------------------------------===//
+// Scalar unary ops for lowering ONNXGeluOp
+//===----------------------------------------------------------------------===//
+template <>
+struct ScalarOp<ONNXGeluOp> {
+  using FOp = CustomScalarOp;
+  using IOp = CustomScalarOp;
+};
+
+template <>
+double analyzeSimdFor<ONNXGeluOp>(
+    Type t, Operation *op, int64_t &von, int64_t &son) {
+  double results;
+  StringRef approximate = dyn_cast<ONNXGeluOp>(op).getApproximate();
+  if (approximate == "none")
+    results = simdAnalysis({GenericOps::ArithmeticGop, GenericOps::DivGop,
+                               GenericOps::MulGop, GenericOps::SqrtGop},
+        {1, 1, 2, 1}, t, von, son);
+  if (approximate == "tanh")
```

Comment:
Use `approximate.equals_insensitive("tanh")` to cover different cases. Please make the change at other locations as well.

Reply:
Hmm but according to the documentation it should only be `tanh` and `none`. If onnx does not support different cases, I do not think we should either or else there will be an error.

Reply:
But if we see reasons to have different variations... I am open to change. Thanks for reviewing!

Reply:
I am fine either way, I see that the GELU verifier will bomb if the string is not `none` or `approximate`. What is done for the other strings? would be good to have a consensus.

Reply:
You are right... whatever is done throughout the code basis is what should be followed. I see it done in other places, so I do not mind adding it but yes I made sure that it will fail if it is not what onnx is expecting :)

Reply:
Okay done!